core/src/test/java/org/apache/iceberg/TestScansAndSchemaEvolution.java

```diff
@@ -113,4 +116,79 @@ public void testPartitionSourceRename() throws IOException {
 
     assertThat(tasks).hasSize(1);
   }
+
+  @TestTemplate
+  public void testAddColumnWithDefaultValueAndQuery() throws IOException {
+    assumeThat(V3_AND_ABOVE).as("Default values require v3+").contains(formatVersion);
+    File location = Files.createTempDirectory(temp, "junit").toFile();
+    assertThat(location.delete()).isTrue(); // should be created by table create
+
+    Table table = TestTables.create(location, "test", SCHEMA, SPEC, formatVersion);
+
+    // Write initial data
+    DataFile fileOne = createDataFile("one");
+    DataFile fileTwo = createDataFile("two");
+    table.newAppend().appendFile(fileOne).appendFile(fileTwo).commit();
+
+    // Add a new column with an initial default value
+    String defaultValue = "default_category";
+    table
+        .updateSchema()
+        .addColumn("category", Types.StringType.get(), "Product category", Literal.of(defaultValue))
+        .commit();
+
+    // Verify the schema includes the new column with default value
+    Schema updatedSchema = table.schema();
+    Types.NestedField categoryField = updatedSchema.findField("category");
+    assertThat(categoryField).isNotNull();
+    assertThat(categoryField.initialDefault()).isEqualTo(defaultValue);
+    assertThat(categoryField.writeDefault()).isEqualTo(defaultValue);
+
+    // Verify scan planning works with the new column that has default value
+    assertThat(table.newScan().planFiles()).hasSize(2);
+
+    // Test that scan with projection includes the new column with default value
+    Schema projectionSchema = table.schema().select("id", "data", "category");
+    List<FileScanTask> projectionTasks =
+        Lists.newArrayList(table.newScan().project(projectionSchema).planFiles());
+    assertThat(projectionTasks).hasSize(2);
+
+    // Verify that each task has the correct schema with the default column
+    for (FileScanTask task : projectionTasks) {
```

Comment:
btw you could also use `allSatisfy()` here as well

Reply:
Done